From 339b815053ceeed0c1b57f8c8f7d5c502f2ce12d Mon Sep 17 00:00:00 2001
From: James Page <james.page@chainguard.dev>
Date: Tue, 12 Aug 2025 13:10:23 +0000
Subject: [PATCH] CustomSocketFactory: use default Keystore type

Fallback to the default Keystore type rather than assuming that this is
"JKS"; when deployed with Bouncycastle FIPS this might be "BCFKS".
---
 .../org/apache/hop/core/truststore/CustomSocketFactory.java     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/engine/src/main/java/org/apache/hop/core/truststore/CustomSocketFactory.java b/engine/src/main/java/org/apache/hop/core/truststore/CustomSocketFactory.java
index fc8234c4cf..395377cc60 100644
--- a/engine/src/main/java/org/apache/hop/core/truststore/CustomSocketFactory.java
+++ b/engine/src/main/java/org/apache/hop/core/truststore/CustomSocketFactory.java
@@ -80,7 +80,7 @@ public class CustomSocketFactory extends SSLSocketFactory {
       if (!Utils.isEmpty(path) && path.endsWith(".p12")) {
         keyStore = KeyStore.getInstance("PKCS12");
       } else {
-        keyStore = KeyStore.getInstance("JKS");
+        keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
       }
     } catch (Exception e) {
       throw new HopException(
-- 
2.50.1

